desktop-widgets/importgps.cpp

```diff
@@ -0,0 +1,329 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "ui_importgps.h"
+#include "desktop-widgets/importgps.h"
+#include "desktop-widgets/locationinformation.h"
+#include "core/dive.h"
+
+#include <time.h>
+#include <QFileDialog>
+#include <QProcess>
+#include <QDateTime>
+#include <QDate>
+#include <stdlib.h>
+#include <stdio.h>
+#include <QDebug>
+
+#define BUFFER_SIZE 5000
+
+/* Import dive coordinates from a GPS device and synchronise them with the dive profile information
+   of a dive computer. This file contains the infrastructure to:
+   1) Read a .GPX file from a GPS system.
+   2) Find the first gpx trackpoint that follows after the start of the dive.
+   3) Allow modification of the coordinates dependent on international time zone and
+              on differences in local time settings between the GPS and the dive computer.
+   4) Saving the coordinates into the Coordinates text box in the Dive Site Edit panel
+              and which which causes the map to show the location of the dive site. 
+   The structure coords is used to store critical information.  */
+ImportGPS::ImportGPS(QWidget *parent, QString fileName, class Ui::LocationInformation *LocationUI) : QDialog(parent),
+	fileName(fileName), LocationUI(LocationUI)
+{
+	ui.setupUi(this);
+	connect(ui.timeDiffEdit, SIGNAL(timeChanged(const QTime)), this, SLOT(timeDiffEditChanged()));
+	connect(ui.timeZoneEdit, SIGNAL(timeChanged(const QTime)), this, SLOT(timeZoneEditChanged()));
+	connect(ui.timezone_backwards, SIGNAL(toggled(bool)), this, SLOT(changeZoneBackwards()));
+	connect(ui.timezone_forward, SIGNAL(toggled(bool)), this, SLOT(changeZoneForward()));
+	connect(ui.diff_backwards, SIGNAL(toggled(bool)), this, SLOT(changeDiffBackwards()));
+	connect(ui.diff_forward, SIGNAL(toggled(bool)), this, SLOT(changeDiffForward()));
+	connect(ui.GPSbuttonBox, SIGNAL(clicked(QAbstractButton *)), this, SLOT(buttonClicked(QAbstractButton *)));
+	coords.settingsDiff_offset = 0;
+	coords.timeZone_offset = 0;
+	coords.lat = 0;
+	coords.lon = 0;
+	pixmapSize = (int) (ui.diveDateLabel->height() / 2);
+}
+
+void  ImportGPS::buttonClicked(QAbstractButton *button)
+{
+	if (ui.GPSbuttonBox->buttonRole(button) == QDialogButtonBox::AcceptRole) {
+		// Write the coordinates in decimal degree format to the Coordinates QLineEdit of the LocationaInformationWidget UI:
+		LocationUI->diveSiteCoordinates->setText(QString::number(coords.lat) + ", " + QString::number(coords.lon));
+		LocationUI->diveSiteCoordinates->editingFinished();
+	} else {
+		close();
+	}
+}
+
+// Extract text from the present position in the file until
+//  the first 'delim' character is encountered.
+int ImportGPS::get_substring(FILE *f, char *buf, char delim)
```

Comment:
is this a naming convention that has settled?
should all C++ member functions be `pascalCase` e.g. `ImportGPS::getSubstring`?

Reply:
Where's the user of that function?
In C++ code one might use QString or std::string instead of fixed-size buffers and all that brittle stuff.

Reply:
Buffer converted to QString.